agent/agent.go

```diff
@@ -91,35 +91,36 @@ func (a *Agent) Run(payload *queue.Work, cancel <-chan bool) error {
 }
 
 func (a *Agent) prep(w *queue.Work) (*yaml.Config, error) {
-
 	envs := toEnv(w)
 	w.Yaml = expander.ExpandString(w.Yaml, envs)
 
-	// inject the netrc file into the clone plugin if the repository is
-	// private and requires authentication.
+	// append all secrets. they will be checked by the ImageSecrets transform.
 	var secrets []*model.Secret
-	if w.Verified {
-		secrets = append(secrets, w.Secrets...)
-	}
+	secrets = append(secrets, w.Secrets...)
```

Comment:
The behavior @tboerger is describing is also true for org secrets. If the org is skip and repo is not then the org one will be used.

If this is a problem then the secrets can be merged in the function that populates the list.